imap_processing/ialirt/process_ephemeris.py

```diff
@@ -158,6 +135,66 @@ def calculate_azimuth_and_elevation(
     return np.asarray(azimuth), np.asarray(elevation)
 
 
+@typing.no_type_check
+def calculate_doppler(
+    longitude: float,
+    latitude: float,
+    altitude: float,
+    observation_time: Union[float, np.ndarray],
+    target: str = SpiceBody.IMAP.name,
+) -> Union[float, ndarray[float]]:
+    """
+    Calculate the doppler velocity.
+
+    Notes about the spkezr function:
+    The function returns the state of the target (state) and the light time.
+    The first three components of state represent the x-, y- and z-components of the
+    target's position; the last three components form the corresponding velocity vector.
+
+    Parameters
+    ----------
+        longitude : float
+        Longitude in decimal degrees. Positive east of prime meridian,
+        negative to west.
+    latitude : float
+        Latitude in decimal degrees. Positive north of equator, negative
+        to south.
+    altitude : float
+        Altitude in kilometers.
+    observation_time : float or np.ndarray
+        Time at which the state of the target relative to the observer
+        is to be computed. Expressed as ephemeris time, seconds past J2000 TDB.
+    target : str (Optional)
+        The target body. Default is "IMAP".
+
+    Returns
+    -------
+    doppler : float or np.ndarray[float]
+        Doppler velocity in kilometers per second.
+    """
+    ground_station_position_ecef = latitude_longitude_to_ecef(
+        longitude, latitude, altitude
+    )
+
+    if not isinstance(observation_time, np.ndarray):
+        observation_time = [observation_time]
+
+    # TODO: consider improving vectorizing this for loop by moving everything besides
+    #  the spkezr outside of the for loop,
+    # TODO: which would require the axis argument
```

Comment:
Open to thoughts here about whether or not I should move the non-spkezr lines out of the for loop below, and instead append the state, light_time results to an array that gets modified in these moved lines outside of the loop. This might improve performance slightly, but I did not make this change because this felt simpler to me.

Reply:
Actually, I don't think that you need any for loop as it looks like `spiceypy.spkezr` is already vectorized.